### PR TITLE
added AWD support (2x X, 2x Y)

### PR DESCRIPTION
**Pull request by Surion79**
this adds additional config possibilities for "AWD" or multi driver per axis X and Y. (max 2 per axis)
It is implemented like the Z implementation.

For not breaking, the original files reference to the "1-Motor" variant of the new files.

### CODE REVIEW

**Comment by Frix-x** [comment]
Thanks for the PR!

I think it's giving a lot of changes and there is already a lot ready for the new release. So if you don't mind, I'll merge it later when you get your motors and everything to test it properly :)

**Comment by Surion79** [comment]
I'm fine with it. I do need it merged in develop at 18th march 2024. then i habe the last parts delivered. Motors are inabout next week (ldo nema14 high temp).
I won't be able to test all tmcs since i don't own 5160s. since it is derived from the newest develop version, it should have the correct parameters.

**Comment by Kcdc3d** [comment]
Hi there, I am trying to figure out if the AWD_XY branch is usable or if I should wait until this is merged to Main... I have the AWD-330 vzbot and would like to try out your suite when it's ready.

**Comment by Surion79** [comment]
due to personal stuff i wasn't able to test it out, although there is no coding per se involved. it is just configuration files for the additional X/Y motors.

**Comment by Kcdc3d** [comment]
Ok I thought I saw a few files changed but after going back, it was just another driver setup for the second motor. Thanks for clearing that up!

**Comment by tristanmcpherson** [comment]
Testing this now, we will also need a replacement for the default wiring imports in `[include config/hardware/axis/X/1.8deg.cfg]` so possible alongside `[include config/hardware/axis/X/1.8deg-2_Motors.cfg]`, etc?

**Comment by Surion79** [comment]
what do you mean? I found a missing config regarding default speeds. but wiring? It is 1M, and 2M

**Comment by Surion79** [comment]
let me have a look

**Comment by Surion79** [comment]
@tristanmcpherson mind testing again with the new addition? or is there something different missing as well?
It is not possible to add templates like Z axis, as there is not really a typical AWD only builds for referencing

**Comment by tristanmcpherson** [comment]
> @tristanmcpherson mind testing again with the new addition? or is there something different missing as well? It is not possible to add templates like Z axis, as there is not really a typical AWD only builds for referencing

Working good with the new changes. Was the difference in file name for 0.9deg intentional? The space seems potentially problematic

**Comment by Surion79** [comment]
well, copy paste topic :D anything else? :)

**Comment by tristanmcpherson** [comment]
Working good for me! I can confirm good enough to be merged @Frix-x

**Comment by tristanmcpherson** [comment]
Oh, we should add user template for the new motor configs in https://github.com/Frix-x/klippain/blob/main/user_templates/printer.cfg#L27

**Comment by Surion79** [comment]
added, anything else? :)

**Comment by Kcdc3d** [comment]
cheers and thanks for the updates

File: user_templates/printer.cfg
Comment (Frix-x) [inline user_templates/printer.cfg:35]: For each line, can you remove the word "Voron" here? This line is for standard 1 motor axis, whatever the printer brand or type.
Second line is for AWD (2 motors for X) and this is ok like this.

**Review by Frix-x — CHANGES_REQUESTED** [review]
I'm almost ready to merge it. Just a small change in the comments added in the printer.cfg are requested to generalize a bit more the lines

File: user_templates/printer.cfg
Comment (Surion79) [reply user_templates/printer.cfg]: removed it

**Comment by Surion79** [comment]
i removed the whole "e.g." parts to make it more generic

**Review by Frix-x — APPROVED** [review]
Ok that's good now

**Pull request merged**
